src/noteburst/dependencies/arqpool.py

```diff
@@ -0,0 +1,461 @@
+"""A FastAPI dependency that supplies a Redis connection for arq."""
```

Comment:
I'm of two minds about including all the models directly in the dependency. I had previously been leaning towards separating out the models and putting them in a file under models because that matches what we usually do in FastAPI applications, but I can't think of a strong argument for that being better necessarily. But we should probably decide how we want to handle this as we import more dependencies into Safir and have to decide whether all the models should be imported from the dependency or from `safir.models`.

Reply:
What models are you thinking of? In arqpool I've got a couple dataclasses but those aren't models suitable for an HTTP API. They're really just for the internal usage of the arq queue. Then the `QueuedJob` model associated with the `prototype` API takes those dataclasses and transforms them into the public-facing API.

In otherwords, I do like the architecture I've got here, where the models associated with the usage of the dependency are in the same module as the dependency. The public HTTP API is another thing entirely, and how jobs should be represented in the API aren't part of the dependency codebase.

I like the idea of internal models being separate from public models.

Now maybe the equivalent of `QueuedJob` model should also get lifted into Safir? I don't know, to be honest. Perhaps if we want Safir to provide a full set of endpoints for queue management? But I'm also a little reluctant to put that into Safir now because different apps may need to implement their queued job metatadata APIs differently and standardizing up-front might be counterproductive.

Reply:
Oh, sorry, my terminology was unnecessarily confusing. I tend to separate the data classes from the code at all levels, not just when they're response models, because that's a hexagonal architecture thing. But in this case I agree that it doesn't serve any real point and it makes more sense to keep the exceptions and dataclasses in the same module as the implementation.